ooniprobe/Utility/TestLists.mm

```diff
@@ -0,0 +1,76 @@
+#import "TestLists.h"
+#import "NTYCSVTable.h"
+#include <measurement_kit/ooni.hpp>
+#include <measurement_kit/ooni/orchestrate.hpp>
+
+@implementation TestLists
+
++ (id)sharedTestLists
+{
+    static TestLists *sharedTestLists = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        sharedTestLists = [[self alloc] init];
+    });
+    return sharedTestLists;
+}
+
+-(id)init
+{
+    self = [super init];
+    if(self)
+    {
+        [self getCC];
+    }
+    return self;
+}
+
+- (NSArray*)getUrls{
+    NSMutableArray *urls = [[NSMutableArray alloc] init];
+    NSArray *global_urls = [self getUrlsForCountry:@"global"];
+    [urls addObjectsFromArray:global_urls];
+    NSArray *local_urls = [self getUrlsForCountry:[self.probe_cc lowercaseString]];
+    if (local_urls != nil)
+        [urls addObjectsFromArray:local_urls];
+    //NSLog(@"url %@", urls);
+    return urls;
+}
+
+- (NSArray*)getUrlsForCountry:(NSString*)country_code {
+    if ([[NSBundle mainBundle] pathForResource:country_code ofType:@"csv"] != nil){
+        NSURL *csvURL = [NSURL URLWithString:[NSString stringWithFormat:@"file://%@", [[NSBundle mainBundle] pathForResource:country_code ofType:@"csv"]]];
+        NTYCSVTable *table = [[NTYCSVTable alloc] initWithContentsOfURL:csvURL];
+        NSArray *url = table.columns[@"url"];
+        return url;
+    }
+    return nil;
+}
+
+- (void)getCC {
+    //TODO refactor all mk helper functions into a class
+    //This will be done with the next release as this class is going to be deleted
+    NSBundle *bundle = [NSBundle mainBundle];
+    NSString *geoip_asn_path = [bundle pathForResource:@"GeoIPASNum" ofType:@"dat"];
+    NSString *geoip_country_path = [bundle pathForResource:@"GeoIP" ofType:@"dat"];
+    mk::ooni::orchestrate::Client client;
```

Comment:
I think here you may want to refactor the code to call the free function `find_location`.

Reply:
done